packages/components/src/HeaderBar/HeaderBar.component.js

```diff
@@ -231,20 +230,20 @@ function HeaderBar(props) {
 		User,
 		Help,
 		Products,
-	}, props.renderers || {});
+	}, getComponents(renderers) || {});
 
 	return (
 		<nav className={classNames(theme['tc-header-bar'], 'tc-header-bar')}>
 			<ul className={theme['tc-header-bar-actions']}>
-				<Components.Logo renderers={renderers} {...props.logo} />
-				<Components.Brand renderers={renderers} {...props.brand} isSeparated={!!props.env} />
-				{ props.env && <Components.Environment renderers={renderers} {...props.env} /> }
+				{ logo && <Components.Logo {...{ Components }} {...logo} />}
+				{ brand && <Components.Brand {...{ Components }} isSeparated={!!env} {...brand} />}
+				{ env && <Components.Environment {...{ Components }} {...env} /> }
 			</ul>
 			<ul className={classNames(theme['tc-header-bar-actions'], theme.right)}>
-				{ props.search && <Components.Search renderers={renderers} {...props.search} /> }
-				{ props.help && <Components.Help renderers={renderers} {...props.help} /> }
-				{ props.user && <Components.User renderers={renderers} {...props.user} /> }
-				{ props.products && <Components.Products renderers={renderers} {...props.products} /> }
+				{ search && <Components.Search {...{ Components }} {...search} /> }
```

Comment:
I find the `{...{ Components }}` syntax confusing.
Do you think it could be improved.